<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AY" />


<title>P135</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Jacque B. Projects - Moeller lab</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="P95.html">P95</a>
</li>
<li>
  <a href="GSE101463.html">GSE101463</a>
</li>
<li>
  <a href="GSE75060.html">GSE75060</a>
</li>
<li>
  <a href="P135.html">P135</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="contact.html">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://github.com/frymor">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/AssaYeroslaviz">
    <span class="fa fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/ayeroslaviz/">
    <span class="fa fa-linkedin fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">P135</h1>
<h4 class="author">AY</h4>
<h4 class="date">11/4/2019</h4>

</div>


<div id="analysis-fastq-files-for-project-p135" class="section level2">
<h2>Analysis fastq files for project P135</h2>
<p>by the look of the data it is a paired-end sample dataset. Jacque asked me to amp it both to the Dm6 and the Dp3 genomes.</p>
<p>“The only requirement in term of bioinformatic analysis is the mapping of the reads on the dm6 genome + the D. pseudoobscura dp3 genome (pseudoobscura chromatin was spiked in the melanogaster chromatin for downstream normalisation)”</p>
<div id="mapping-to-dm6" class="section level3">
<h3>Mapping to dm6</h3>
<div id="generate-genomic-index-for-star" class="section level4">
<h4>Generate genomic index for STAR</h4>
<pre class="bash"><code>cd /home/yeroslaviz/genomes/Drosophila_melanogaster/Dm6

STAR --runMode genomeGenerate --runThreadN 15 --genomeDir STARIndex/ --genomeFastaFiles Drosophila_melanogaster.BDGP6.22.dna.toplevel.fa --sjdbGTFfile Drosophila_melanogaster.BDGP6.22.45.gtf  sjdbOverhang 99</code></pre>
<p>the samples here should be mapped to genome build <em>dm6</em>.</p>
<pre class="bash"><code>
cd /home/yeroslaviz/projects/JacqueB/P135/


starIndex=&quot;/home/yeroslaviz/genomes/Drosophila_melanogaster/Dm6/STARIndex&quot;

mkdir -p /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dm6_STARMapping/ # keep permissions

for file in `ls -1 rawData/*.R1.fastq.gz`
do
#echo &quot;$file&quot;
base=$(echo $file | sed -E &quot;s/rawData\/(.*).R1.fastq.gz/\1/&quot;)
#base=$(echo $file | sed -E &quot;s/(.*)\_1\.fastq/\1/&quot;)
echo &quot;running STAR for file $base&quot;
echo &quot;STAR --genomeDir $starIndex --readFilesIn rawData/$base\.R1.fastq.gz rawData/$base\.R2.fastq.gz --outFileNamePrefix Analysis/Dm6_STARMapping/$base&quot;
STAR --runThreadN 15 --outBAMsortingThreadN 10 --outReadsUnmapped Fastx --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --genomeDir $starIndex --outFileNamePrefix Analysis/Dm6_STARMapping/$base --readFilesIn rawData/$base\.R1.fastq.gz rawData/$base\.R2.fastq.gz
done</code></pre>
</div>
</div>
</div>
<div id="indexing" class="section level2">
<h2>indexing</h2>
<pre class="bash"><code>cd /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dm6_STARMapping/
  
for i in ` ls -1 *.bam`
do
echo $i
samtools index $i
done</code></pre>
<div id="convet-bam-2-bigwig" class="section level4">
<h4>convet bam 2 bigwig</h4>
<div id="get-indexed-fasta-file" class="section level5">
<h5>get indexed fastA file</h5>
<pre class="bash"><code>samtools faidx /home/yeroslaviz/genomes/Drosophila_melanogaster/Dm6/Drosophila_melanogaster.BDGP6.22.dna.toplevel.fa
cut -f 1,2  /home/yeroslaviz/genomes/Drosophila_melanogaster/Dm6/Drosophila_melanogaster.BDGP6.22.dna.toplevel.fa.fai &gt; Dm6.chromSize</code></pre>
</div>
<div id="convert-bam-to-wig-to-bigwig" class="section level5">
<h5>convert bam to wig to bigwig</h5>
<pre class="bash"><code>mkdir -p /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dm6_bigwig/

cd /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dm6_STARMapping/
for file in ` ls -1 *.bam`
do
base=$(echo $file | sed -E &quot;s/.concAligned.sortedByCoord.out.bam//&quot;)
echo $base
bam2wig.py -i $file -s Dm6.chromSize -o /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dm6_bigwig/$base &amp;&gt; /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dm6_bigwig/$base.wigFiles.log
rm -f /home/yeroslaviz/projects/JacqueB/P135/Anaysis/Dm6_bigwig/$base.wig
done</code></pre>
<div id="section" class="section level38">
<p></p>
</div>
<div id="section-1" class="section level38">
<p></p>
</div>
<div id="section-2" class="section level38">
<p></p>
</div>
</div>
</div>
<div id="mapping-to-dp3" class="section level3">
<h3>Mapping to Dp3</h3>
<div id="generate-genomic-index-for-star-1" class="section level4">
<h4>Generate genomic index for STAR</h4>
<p>As this is the first time, the indexed genome must be created first.</p>
<pre class="bash"><code>cd /home/yeroslaviz/genomes/Drosophila_pseudoobscura/Dp3

STAR --runMode genomeGenerate --runThreadN 15 --genomeDir STARIndex/ --genomeFastaFiles Drosophila_pseudoobscura.Dpse_3.0.dna.toplevel.fa --sjdbGTFfile Drosophila_pseudoobscura.Dpse_3.0.45.gtf  sjdbOverhang 99</code></pre>
<p>the samples here should be mapped to genome build <em>Dp3</em>.</p>
<pre class="bash"><code>
cd /home/yeroslaviz/projects/JacqueB/P135

starIndex=&quot;/home/yeroslaviz/genomes/Drosophila_pseudoobscura/Dp3/STARIndex&quot;

mkdir -p /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dp3_STARMapping/ # keep permissions

for file in `ls -1 rawData/*.R1.fastq.gz`
do
#echo &quot;$file&quot;
base=$(echo $file | sed -E &quot;s/rawData\/(.*).R1.fastq.gz/\1/&quot;)
#base=$(echo $file | sed -E &quot;s/(.*)\_1\.fastq/\1/&quot;)
echo &quot;running STAR for file $base&quot;
echo &quot;STAR --genomeDir $starIndex --readFilesIn rawData/$base\.R1.fastq.gz rawData/$base\.R2.fastq.gz --outFileNamePrefix Analysis/Dp3_STARMapping/$base&quot;
STAR --runThreadN 5 --outBAMsortingThreadN 5 --outReadsUnmapped Fastx --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --genomeDir $starIndex --outFileNamePrefix Analysis/Dp3_STARMapping/$base --readFilesIn rawData/$base\.R1.fastq.gz rawData/$base\.R2.fastq.gz
done</code></pre>
</div>
</div>
</div>
<div id="indexing-1" class="section level2">
<h2>indexing</h2>
<pre class="bash"><code>cd /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dp3_STARMapping/
  
for i in ` ls -1 *.bam`
do
echo $i
samtools index $i
done</code></pre>
<div id="convet-bam-2-bigwig-1" class="section level4">
<h4>convet bam 2 bigwig</h4>
<div id="get-indexed-fasta-file-1" class="section level5">
<h5>get indexed fastA file</h5>
<pre class="bash"><code>samtools faidx /home/yeroslaviz/genomes/Drosophila_pseudoobscura/Dp3/Drosophila_pseudoobscura.Dpse_3.0.dna.toplevel.fa
cut -f 1,2 /home/yeroslaviz/genomes/Drosophila_pseudoobscura/Dp3/Drosophila_pseudoobscura.Dpse_3.0.dna.toplevel.fa.fai &gt; Dp3.chromSize</code></pre>
</div>
<div id="convert-bam-to-wig-to-bigwig-1" class="section level5">
<h5>convert bam to wig to bigwig</h5>
<pre class="bash"><code>mkdir -p /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dp3_bigwig/

cd /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dp3_STARMapping/
for file in ` ls -1 *.bam`
do
base=$(echo $file | sed -E &quot;s/.concAligned.sortedByCoord.out.bam//&quot;)
echo $base
bam2wig.py -i $file -s Dp3.chromSize -o /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dp3_bigwig/$base &amp;&gt; /home/yeroslaviz/projects/JacqueB/P135/Analysis/Dp3_bigwig/$base.wigFiles.log
rm -f /home/yeroslaviz/projects/JacqueB/P135/Anaysis/Dp3_bigwig/$base.wig

done</code></pre>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
