<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AY" />


<title>P59</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Jacque B. Projects - Moeller lab</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="GSE101463.html">GSE101463</a>
</li>
<li>
  <a href="GSE75060.html">GSE75060</a>
</li>
<li>
  <a href="P135.html">P135</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="contact.html">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://github.com/frymor">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/AssaYeroslaviz">
    <span class="fa fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/ayeroslaviz/">
    <span class="fa fa-linkedin fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">P59</h1>
<h4 class="author">AY</h4>
<h4 class="date">11/6/2019</h4>

</div>


<p>list of files: dros-ChIP_S13_L001_R1_001.fastq dros-ChIP_S13_L001_R2_001.fastq dros-ChIP_S13_L002_R1_001.fastq dros-ChIP_S13_L002_R2_001.fastq dros-ChIP_S13_L003_R1_001.fastq dros-ChIP_S13_L003_R2_001.fastq dros-ChIP_S13_L004_R1_001.fastq dros-ChIP_S13_L004_R2_001.fastq</p>
<div id="concatenate-the-r1r2-files" class="section level3">
<h3>concatenate the R1/R2 files</h3>
<p>cat dros-ChIP_S13_<em><em>R1_001.fastq &gt; dros-ChIP_S13_R1.fastq cat dros-ChIP_S13</em></em>_R2_001.fastq &gt; dros-ChIP_S13_R2.fastq</p>
</div>
<div id="run-fastqc-on-the-samples" class="section level3">
<h3>run fastqc on the samples</h3>
<p>fastqc -t 2 dros-ChIP_S13_R*</p>
</div>
<div id="run-fastx_collapse-on-the-data" class="section level3">
<h3>run fastx_collapse on the data</h3>
<p>for i in <code>find . -name &quot;dros-ChIP_S13_R*.fastq&quot;</code> do base=$(basename <span class="math inline">\(i .fastq) echo &quot;\)</span>base&quot; fastx_collapser -i $i -o $base.collapsed.fastq -v -Q33 done</p>
<p>dros-ChIP_S13_R2 Input: 21068794 sequences (representing 21068794 reads) Output: 4921127 sequences (representing 21068794 reads) dros-ChIP_S13_R1 Input: 21068794 sequences (representing 21068794 reads) Output: 4117542 sequences (representing 21068794 reads)</p>
</div>
<div id="finding-how-many-of-the-collapsed-files-are-unique" class="section level1">
<h1>finding how many of the collapsed files are unique</h1>
<p>grep -c &quot;&gt;*-1<span class="math inline">\(&quot; dros-ChIP_S13_R1.collapsed.fastq 1436939 grep -c &quot;&gt;*-1\)</span>&quot; dros-ChIP_S13_R2.collapsed.fastq 2276451</p>
<div id="run-trim_galore-to-filter-out-low-quality-reads" class="section level3">
<h3>run trim_galore to filter out low quality reads</h3>
<p>trim_galore -q 20 –fastqc –gzip –length 20 -o trimGalore.dros-ChIP_S13/ –paired –retain_unpaired -r1 35 -r2 35 –phred33 dros-ChIP_S13_R1.fastq dros-ChIP_S13_R2.fastq &gt; trimgaloreOut.dros-ChIP_S13.txt 2&gt;&amp;1</p>
</div>
<div id="run-fastx_trimmer-to-trim-the-first-5-and-the-last-base-of-the-reads-for-better-mappability." class="section level3">
<h3>run fastx_trimmer to trim the first 5 and the last base of the reads for better mappability.</h3>
<p>for i in <code>find . -name &quot;dros-ChIP_S13_R*.fastq&quot;</code> do base=$(basename <span class="math inline">\(i .fastq) echo &quot;\)</span>base&quot; fastx_trimmer -f 6 -l 75 -i $i -o $base.trimmed.fastq -v -Q33 done</p>
</div>
<div id="download-the-drosophila-genome-build-dm6-ucscufrom-igenomes" class="section level3">
<h3>download the Drosophila genome build dm6 (UCSCU)from iGenomes</h3>
</div>
<div id="map-the-fastq-files-pe-to-the-drosophila-genome-using-star" class="section level3">
<h3>map the fastq files (PE) to the drosophila genome using STAR</h3>
</div>
</div>
<div id="generate-the-genome-index" class="section level1">
<h1>generate the genome index</h1>
<p>/home/yeroslaviz/software/STAR-STAR-2.5.1a/source/STAR –runThreadN 10 –runMode genomeGenerate –genomeDir STARIndex/ –genomeFastaFiles WholeGenomeFasta/genome.fa</p>
</div>
<div id="mapping-the-raw-fastq-files" class="section level1">
<h1>mapping the raw fastq files</h1>
<p>/home/yeroslaviz/software/STAR-2.5.2b/bin/Linux_x86_64/STAR –runThreadN 10 –genomeDir ~/genomes/Drosophila_melanogaster/UCSC/dm6/Sequence/STARIndex/ –readFilesIn ../rawData/dros-ChIP_S13_R1.fastq ../rawData/dros-ChIP_S13_R2.fastq –outFileNamePrefix dros-ChIP_S13.rawData. –outFilterMultimapNmax 1 –alignIntronMax 1 –outSAMtype BAM SortedByCoordinate –outWigType wiggle read1_5p –outReadsUnmapped Fastx –alignEndsType EndToEnd</p>
<p>samtools index dros-ChIP_S13.rawData.Aligned.sortedByCoord.out.bam</p>
</div>
<div id="mapping-the-trimmed-fastq-files-fastx_trimmer" class="section level1">
<h1>mapping the trimmed fastq files (fastx_trimmer)</h1>
<p>/home/yeroslaviz/software/STAR-2.5.2b/bin/Linux_x86_64/STAR –runThreadN 10 –genomeDir ~/genomes/Drosophila_melanogaster/UCSC/dm6/Sequence/STARIndex/ –readFilesIn ../trimmed.fastq/dros-ChIP_S13_R1.trimmed.fastq ../trimmed.fastq/dros-ChIP_S13_R2.trimmed.fastq –outFileNamePrefix fastx_trimmer_mapping/dros-ChIP_S13.trimmed. –outFilterMultimapNmax 1 –alignIntronMax 1 –outSAMtype BAM SortedByCoordinate –outWigType wiggle read1_5p –outReadsUnmapped Fastx –alignEndsType EndToEnd</p>
<p>samtools index dros-ChIP_S13.trimGalore.Aligned.sortedByCoord.out.bam</p>
</div>
<div id="mapping-the-filtered-fastq-files-trim_galore" class="section level1">
<h1>mapping the filtered fastq files (trim_galore)</h1>
<p>/home/yeroslaviz/software/STAR-2.5.2b/bin/Linux_x86_64/STAR –runThreadN 10 –readFilesCommand zcat –genomeDir ~/genomes/Drosophila_melanogaster/UCSC/dm6/Sequence/STARIndex/ –readFilesIn ../trimGalore.dros-ChIP_S13/dros-ChIP_S13_R1_val_1.fq.gz ../trimGalore.dros-ChIP_S13/dros-ChIP_S13_R2_val_2.fq.gz –outFileNamePrefix trimGalore_mapping/dros-ChIP_S13.trimGalore. –outFilterMultimapNmax 1 –alignIntronMax 1 –outSAMtype BAM SortedByCoordinate –outWigType wiggle read1_5p –outReadsUnmapped Fastx –alignEndsType EndToEnd</p>
<p>samtools index dros-ChIP_S13.trimmed.Aligned.sortedByCoord.out.bam</p>
</div>
<div id="convert-the-bam-files-into-bigwig-for-better-visualisation" class="section level1">
<h1>convert the bam files into bigwig for better visualisation</h1>
<p>for i in <code>ls -1 *.bam</code> do prefix=$(basename $i .bam) echo “running bam2wig.py for $prefix” bam2wig.py -i $i -s dm6.genome -o $prefix echo “running bamTobed–bedGraphToBigWig for $prefix” bamToBed -i $i | genomeCoverageBed -i stdin -bg -g dm6.genome | bedClip stdin dm6.genome $prefix.bg.clip bedGraphToBigWig $prefix.bg.clip dm6.genome $prefix.bw done</p>
<div id="mapping-the-raw-data-files-with-bowtie1-and-converting-them-to-bigwig" class="section level3">
<h3>mapping the raw data files with bowtie1 and converting them to bigWig</h3>
<pre><code>bowtie -p 10 -m 1 -S --un $NEW_FILE.unmapped -q ~/genomes/Drosophila_melanogaster/UCSC/dm6/Sequence/BowtieIndex/genome -1 rawData/dros-ChIP_S13_R1.fastq -2 rawData/dros-ChIP_S13_R2.fastq bowtie1Mapping/dros-ChIP_S13_R1.rawData.sam 2&gt; bowtie1Mapping/dros-ChIP_S13_R1.rawData.stat
echo &quot;sorting and indexing $NEW_FILE&quot;
samtools view -Shu bowtie1Mapping/dros-ChIP_S13_R1.rawData.sam&gt; bowtie1Mapping/dros-ChIP_S13_R1.rawData.bam
samtools sort bowtie1Mapping/dros-ChIP_S13_R1.rawData.bam bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted
samtools index bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bam
rm bowtie1Mapping/dros-ChIP_S13_R1.rawData.sam bowtie1Mapping/dros-ChIP_S13_R1.rawData.bam

echo &quot;running bam2wig.py&quot;
bam2wig.py -i bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bam -s dm6.genome -o bowtie1Mapping/dros-ChIP_S13_R1.bam2wig.rawData

echo &quot;running bam2wig.devonRyan&quot;
bam2wig.devonR  bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bam bowtie1Mapping/dros-ChIP_S13_R1.devonR.rawData
wigToBigWig bowtie1Mapping/dros-ChIP_S13_R1.devonR.rawData.plus.wig dm6.genome bowtie1Mapping/dros-ChIP_S13_R1.devonR.rawData
wigToBigWig bowtie1Mapping/dros-ChIP_S13_R1.devonR.rawData.minus.wig dm6.genome bowtie1Mapping/dros-ChIP_S13_R1.devonR.rawData

echo &quot;running bamTobed--bedGraphToBigWig&quot;
bamToBed -i bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bam | genomeCoverageBed -i stdin -bg -g dm6.genome | bedClip stdin dm6.genome bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bg.clip
bedGraphToBigWig bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bg.clip dm6.genome bowtie1Mapping/dros-ChIP_S13_R1.rawData.sorted.bg.bw</code></pre>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
